### PR TITLE
Fix Cholesky computation for Gelman--Rubin statistic and add per-parameter feature

### PR DESCRIPTION
**Pull request by lukashergt**
There was a little bug in the computation of the Gelman--Rubin statistic. While `np.linalg.cholesky` returns the lower triangular decomposition, `scipy.linalg.cholesky` returns the upper triangular decomposition. Depending on which one we use, we will have to adapt the transposing in matrix multiplications.

This PR also add the feature of returning the Gelman--Rubin statistic _per parameter_, which can sometimes be helpful when assessing convergence and trying to identify the culprit that is difficult to sample.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code is PEP8 compliant (`flake8 anesthetic tests`)
- [x] My code contains compliant docstrings (`pydocstyle --convention=numpy anesthetic`)
- [x] New and existing unit tests pass locally with my changes (`python -m pytest`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have appropriately incremented the [semantic version number](https://semver.org/) in both README.rst and anesthetic/_version.py

### CODE REVIEW

File: anesthetic/samples.py [hunk 1]
```diff
@@ -573,7 +573,7 @@ def remove_burn_in(self, burn_in, reset_index=False, inplace=False):
             data = data.reset_index(drop=True, inplace=inplace)
         return data
 
-    def Gelman_Rubin(self, params=None):
+    def Gelman_Rubin(self, params=None, return_per_param=False):
```
Comment (AdamOrmondroyd) [inline anesthetic/samples.py:576]: Perhaps `per_param` instead of `return_per_param`?

**Review by AdamOrmondroyd — CHANGES_REQUESTED** [review]

**Comment by AdamOrmondroyd** [comment]
Looks good, have just checked that this still runs ok on mac (which was why we changed to scipy cholesky in the first place)

File: anesthetic/samples.py (hunk 1 above)
Comment (lukashergt) [reply anesthetic/samples.py]: Have been contemplating that, too, you suggesting it settles it.

Done in https://github.com/handley-lab/anesthetic/pull/345/commits/299fddb31541ac8aa7e6ca72407b32991661cc6b.

**Review by AdamOrmondroyd — APPROVED** [review]

**Comment by lukashergt** [comment]
Ha, was about to hit "request review", but @AdamOrmondroyd was even faster...^^
Thanks, Adam.

**Comment by AdamOrmondroyd** [comment]
You actually beat me first; I'd made the changes to the docs and hit `git push` only to be met with rejection

I should probably have read the scipy cholesky docs closer before implementing #287

**Review by AdamOrmondroyd — APPROVED** [review]

**Pull request merged**